File: src/ManaFoodProduct.Domain/Entities/Product.cs
namespace ManaFoodProduct.Domain.Entities;

public class Product : BaseEntity
{
    public required string Name { get; set; }
    public string? Description { get; set; }
    public double UnitPrice { get; set; }
    public Guid CategoryId { get; set; }
    public required Category Category { get; set; }
    public List<Item> Items { get; set; } = new();
}

File: src/main/kotlin/br/com/manafood/manafoodpoduct/domain/model/Product.kt
package br.com.manafood.manafoodpoduct.domain.model

import br.com.manafood.manafoodpoduct.domain.common.BaseEntity
import java.time.LocalDateTime
import java.util.UUID

data class Product(
    override val id: UUID,
    override val createdBy: UUID,
    override val createdAt: LocalDateTime = LocalDateTime.now(),
    override val updatedBy: UUID? = null,
    override val updatedAt: LocalDateTime? = null,
    override val deleted: Boolean = false,
    val name: String,
    val description: String,
    val unitPrice: Double,
    val categoryId: UUID,
    val category: Category,
    val items: MutableList<Item> = mutableListOf()
) : BaseEntity(
    id = id,
    createdBy = createdBy,
    createdAt = createdAt,
    updatedBy = updatedBy,
    updatedAt = updatedAt,
    deleted = deleted
)
